stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -196,10 +220,20 @@ public void onHalfClose() {
           @Override
           public void onCancel() {
             responseObserver.cancelled = true;
```

Comment:
We may want to end up disabling the throwing of StatusRuntimeException when trying to send caused by us setting cancelled = true. I in my own code I would want that exception to stop occurring. I'm okay with messing with that later.